database/pgsql/layer.go

```diff
@@ -329,6 +348,21 @@ func (pgSQL *pgSQL) InsertLayer(layer database.Layer) error {
 		}
 	}
 
+	// insert the layer's namespaces
+	stmt, err := tx.Prepare(insertLayerNamespace)
+	for nsid := range namespaceIDs {
+		_, err := stmt.Exec(layer.ID, nsid)
+		if err != nil {
+			tx.Rollback()
+			return handleError("insertLayerNamespace", err)
+		}
+	}
+
+	err = stmt.Close()
```

Comment:
If anything fails while iterating over the namespaceIDs, the statement never gets closed. I think it's best to structure it like this instead:
```go
stmt, err := tx.Prepare(insertLayerNamespace)
defer func() {
  err = stmt.Close()
  if err != nil {
    tx.Rollback()
    log.WithFields(log.Fields{"error": err}).Error("failed to close prepared statement")
}()

for nsid := range namespaceIDs {
  ...
}
```